base/linalg/lq.jl

```diff
@@ -97,17 +97,29 @@ full(Q::LQPackedQ; thin::Bool = true) =
 
 size(A::LQ, dim::Integer) = size(A.factors, dim)
 size(A::LQ) = size(A.factors)
-function size(A::LQPackedQ, dim::Integer)
-    if 0 < dim && dim <= 2
-        return size(A.factors, dim)
-    elseif 0 < dim && dim > 2
+
+# if the original matrix (say A) was square or had more rows than columns,
+# then A's factorization's orthogonal factor (say Q) is a square matrix of
+# order of A's number of columns. if A had more columns than rows, however,
+# then the thin form of Q is the same shape as the original matrix.
+# (in either case, Q's number of columns is that of A's)
+function size(Q::LQPackedQ)
```

Comment:
I believe this should be `size(Q) = (size(Q.factors, 2), size(Q.factors))` to be consistent with QR
```julia
julia> [size(qrfact(randn(n,3))[:Q]) for n in 2:4]
3-element Array{Tuple{Int64,Int64},1}:
 (2, 2)
 (3, 3)
 (4, 4)
```
See also right column of https://github.com/JuliaLang/julia/pull/23729#discussion_r139351887

Reply:
Ah, I thought the factorization was thin by default, and so `size` was supposed to report the thin form's shape?

Reply:
Having read through the equivalent QR code, I believe I now see the overall scheme: `lqfact`/`qrfact` return a thin-form `L`/`R` and an implicit `Q`. That implicit `Q` behaves: (1) like its square form in `size`, in in-place multiplication methods, and in wrappers around LAPACK routines like `ormlq`/`ormqr`; (2) like either its square or thin form in out-of-place multiplication methods dependent upon the shape of other object involved; and (3) like its thin form when converted to an explicit, dense matrix representation.

So IIUC, `size(Q::LQPackedQ)` should return the shape of `Q`'s square form, which will also fix part of #23779 by making `ormlq!`'s checks happy. Please correct me if any of the above misses the mark :). Best!